pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go

```diff
@@ -157,14 +161,25 @@ func (r *ReconcilePerconaServerMongoDBRestore) reconcileRestore(cr *psmdbv1.Perc
 		storageName = bcp.Spec.StorageName
 	}
 
-	pbmc, err := backup.NewPBM(r.client, cr.Spec.ClusterName, cr.Spec.Replset, cr.Namespace)
+	cluster := &psmdbv1.PerconaServerMongoDB{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: cr.Spec.ClusterName, Namespace: cr.Namespace}, cluster)
+	if err != nil {
+		return errors.Wrapf(err, "get cluster %s/%s", cr.Namespace, cr.Spec.ClusterName)
+	}
+
+	pbmc, err := backup.NewPBM(r.client, cluster)
 	if err != nil {
 		return errors.Wrap(err, "create pbm object")
 	}
 	defer pbmc.Close()
 
-	if status.State == psmdbv1.RestoreStateNew {
-		status.PBMname, err = runRestore(bcpName, storageName, pbmc)
+	if status.State == psmdbv1.RestoreStateNew || status.State == psmdbv1.RestoreStateWaiting {
+		stg, ok := cluster.Spec.Backup.Storages[storageName]
+		if !ok {
+			return errors.Errorf("unable to get storage '%s'", cr.Spec.StorageName)
```

Comment:
Since the return parameter (err) is named no need in the explicit assignment before return. `return errors.Errorf...` will be caught by the defer. E.g. https://play.golang.org/p/jI6hucj8_YG